### PR TITLE
Fix Issues with Overlay mode + signUpLink setting on a SPA

### LINKED ISSUE #619: Issues with Overlay mode + signUpLink setting on a SPA
This happens when:

- Lock 10 is being used in a SPA
- Lock 10 is being shown as overlay
- `signUpLink` was set for redirecting the user to a custom signup page

When the user clicks on `sign up`, Lock changes the route but since the SPA router avoids doing the actual redirect, lock is still visible on top of the page content.

It should close itself when it redirects the user

### PR DESCRIPTION
**Pull request by glena**
Closes #619

### CODE REVIEW

File: src/connection/database/login_sign_up_tabs.jsx [hunk 1]
```diff
@@ -2,6 +2,9 @@ import React from 'react';
 import { showLoginActivity, showSignUpActivity } from './actions';
 import * as l from '../../core/index';
 import { getScreen } from './index';
+import {
+  closeLock
```
Comment (hzalaz) [inline src/connection/database/login_sign_up_tabs.jsx]: Not sure what is the style but maybe follow the other import single line style.

Comment (hzalaz) [inline src/connection/database/login_sign_up_tabs.jsx]: @cristiandouce ?

File: src/connection/database/login_sign_up_tabs.jsx [hunk 2]
```diff
@@ -38,9 +42,16 @@ export default class LoginSignUpTabs extends React.Component {
   }
 
   handleSignUpClick() {
+    if (this.props.signUpLink) {
```
Comment (hzalaz) [inline src/connection/database/login_sign_up_tabs.jsx]: Should we consider the `closable` property value with this or should we just ignore it?

File: src/connection/database/login_sign_up_tabs.jsx (hunk 1 above)
Comment (glena) [inline src/connection/database/login_sign_up_tabs.jsx]: right

File: src/connection/database/login_sign_up_tabs.jsx (hunk 2 above)
Comment (glena) [inline src/connection/database/login_sign_up_tabs.jsx]: I don't think so, in this context there are 2 scenarios:

regular webapp: the redirection will change the entire page. It is irrelevant if lock hides or not
spa: changing the route will not force a redirection so lock keeps visible unless it is explicitly hidden (this change).

While implementing lock, you might want it to not be closable (forcing signup or login on a page?) but you will want it to hide if the user is redirected to a different signup page.

**Pull request merged**